users/script.js

```diff
@@ -474,18 +485,92 @@ function getCheckedValues(groupName) {
 
 function getFilteredUsersURL(checkedValuesSkills, checkedValuesAvailability) {
   const params = new URLSearchParams();
-
   checkedValuesSkills.forEach((skill) => {
     params.append('tagId', skill);
   });
-
   checkedValuesAvailability.forEach((availability) => {
     params.append('state', availability);
   });
-
   return `?${params.toString()}`;
 }
 
+function manipulateQueryParamsToURL(constructedQueryParam) {
+  const currentURLInstance = new URL(window.location.href);
+  currentURLInstance.search = '';
+  const currentURL = currentURLInstance.href;
+  if (constructedQueryParam) {
+    const newURLWithQueryParams = `${currentURL}${constructedQueryParam}`;
+    window.history.pushState(
+      { path: newURLWithQueryParams },
+      '',
+      newURLWithQueryParams,
+    );
+  } else {
+    const newUrlWithoutQueryParams = `${currentURL}`;
+    window.history.pushState(
+      { path: newUrlWithoutQueryParams },
+      '',
+      newUrlWithoutQueryParams,
+    );
+  }
+}
+
+function parseQueryParams() {
+  const urlString = window.location.href;
+  const urlObjInstance = new URL(urlString);
+  const queryParamsObj = urlObjInstance.searchParams.entries();
+  const queryObject = {};
+  for (const [key, value] of queryParamsObj) {
+    if (!queryObject[key]) {
+      queryObject[key] = [];
+    }
+    if (key === 'state') {
+      queryObject[key].push(value.toLocaleUpperCase());
+    } else {
+      queryObject[key].push(value);
+    }
+  }
+  return queryObject;
```

Comment:
- Why are we doing this? What will the `queryObject` look like at the end? 
Can you please add a JSDoc comment stating the return type.

Reply:
```
/**
 * Parses the query parameters from the current URL and organizes them into an object.
 *
 * @function
 * @returns {Object.<string, string[]>} An object containing query parameter keys as properties
 * and arrays of corresponding values. Values are either in uppercase (for 'state') or as provided in the URL.
 */
```